hudi-sync/hudi-hive-sync/src/main/java/org/apache/hudi/hive/GlueDataCatalogClient.java

```diff
@@ -0,0 +1,483 @@
+      return (awsGlueClient.getDatabase(request).getDatabase() != null);
+    } catch (EntityNotFoundException exception) {
+      LOG.error("Database " + syncConfig.databaseName, exception);
+    } catch (Exception exception) {
+      LOG.error("Failed to check if database exists " + syncConfig.databaseName, exception);
+      throw new HoodieHiveSyncException("Failed to check if database exists " + syncConfig.databaseName + " in region ", exception);
+    }
+    return false;
+  }
+
+  @Override
+  public void createDatabase() {
+    if (!databaseExists()) {
+      CreateDatabaseRequest request = new CreateDatabaseRequest();
+      request.setDatabaseInput(new DatabaseInput().withName(syncConfig.databaseName).withDescription("automatically created by hudi").withParameters(null).withLocationUri(null));
+      try {
+        CreateDatabaseResult result = awsGlueClient.createDatabase(request);
+        LOG.info("Successfully created database in Glue: " + result.toString());
+      } catch (AlreadyExistsException exception) {
+        LOG.warn("Database " + syncConfig.databaseName + " already exists", exception);
+      } catch (Exception exception) {
+        LOG.error("Failed to create database " + syncConfig.databaseName, exception);
+        throw new HoodieHiveSyncException("Failed to create database " + syncConfig.databaseName, exception);
+      }
+    }
+  }
+
+  @Override
+  public boolean doesTableExist(String tableName) {
+    GetTableRequest request = new GetTableRequest()
+        .withDatabaseName(syncConfig.databaseName)
+        .withName(tableName);
+    try {
+      awsGlueClient.getTable(request);
+      return true;
+    } catch (EntityNotFoundException exception) {
+      LOG.error("Accessing non-existent Glue Table " + tableName + " in database " + syncConfig.databaseName, exception);
+    } catch (Exception exception) {
+      String errorMsg = "Fatal error while fetching Glue Table " + tableName + " in database " + syncConfig.databaseName;
+      LOG.error(errorMsg, exception);
+      throw new HoodieHiveSyncException(errorMsg, exception);
+    }
+    return false;
+  }
+
+  @Override
+  public void createTable(String tableName, MessageType storageSchema,
+                          String inputFormatClass, String outputFormatClass,
+                          String serdeClass, Map<String, String> serdeProperties,
+                          Map<String, String> tableProperties) {
+    if (!doesTableExist(tableName)) {
+      CreateTableRequest request = new CreateTableRequest();
+      Map<String, String> params = new HashMap<>();
+      if (!syncConfig.createManagedTable) {
+        params.put("EXTERNAL", "TRUE");
+      }
+      for (Map.Entry<String, String> entry : tableProperties.entrySet()) {
+        params.put(entry.getKey(), entry.getValue());
+      }
+
+      try {
+        LinkedHashMap<String, String> mapSchema = HiveSchemaUtil.parquetSchemaToMapSchema(storageSchema, syncConfig.supportTimestamp, false);
```

Comment:
Prefer interface over implementation

```suggestion
        Map<String, String> mapSchema = HiveSchemaUtil.parquetSchemaToMapSchema(storageSchema, syncConfig.supportTimestamp, false);
```